core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/data/UserDataBinderImpl.java

```diff
@@ -349,6 +351,15 @@ public PropagationByResource update(final User toBeUpdated, final UserPatch user
 
         if (userPatch.getMustChangePassword() != null) {
             user.setMustChangePassword(userPatch.getMustChangePassword().getValue());
+            Set<ExternalResource> externalResources = anyUtils.getAllResources(toBeUpdated);
```

Comment:
The same code for adding resources to the `propByRes` instance in case such resources contain the corresponding mapping item should also performed in the `create()` method of this class - here being the `update()` method.

Reply:
All resources during the create operation are being added here https://github.com/apache/syncope/blob/85aadb3cd826d34e984540990187552437aa9236/core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/data/AbstractAnyDataBinder.java#L504. Finally the propByRes are populated from saved user entity at this place https://github.com/apache/syncope/blob/85aadb3cd826d34e984540990187552437aa9236/core/workflow-java/src/main/java/org/apache/syncope/core/workflow/java/DefaultUserWorkflowAdapter.java#L85. After this final provisioning into remote storages is happening here https://github.com/apache/syncope/blob/85aadb3cd826d34e984540990187552437aa9236/core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/DefaultUserProvisioningManager.java#L92

I assume that nothing should be added for `create` operation. Am I missing something? Please, correct me if I'm wrong.

Reply:
Note that I'm merging into 2_0_X branch and haven't explored other branches.

Reply:
No, you're right: all resources are already in `propByRes` - resolving this.